app/apollo/resolvers/subscription.js

```diff
@@ -462,19 +466,23 @@ const subscriptionResolvers = {
         const kubeOwnerId = await models.User.getKubeOwnerId(context);
 
         const conditions = await getGroupConditionsIncludingEmpty(me, org_id, ACTIONS.READ, 'name', queryName, context);
-        logger.debug({req_id, user, org_id, conditions }, `${queryName} group conditions are...`);
+        logger.debug({req_id, user, org_id, conditions}, `${queryName} group conditions are...`);
 
         const subscription = await models.Subscription.findOne({ org_id, uuid, ...conditions }, {}).lean({ virtuals: true });
-        if(!subscription){
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - found: ${!!subscription}`);
+
+        const identifiers = subscription ? [uuid, subscription.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.SUBSCRIPTION, queryName, context, identifiers);
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);
+
```

Comment:
Shouldn't read access to to the passed config (channel) and groups also be verified?

Reply:
fixed